### PR TITLE
Add check-modules (true) and check-vendor (false)

### PR DESCRIPTION
**Pull request by ghost**
`check-modules` (defaults to true) runs go mod {tidy,verify} and checks to see if git is
still clean. If it's not then it's an error.

`check-vendor` (defaults to false) runs go mod vendor and checks to see if git is still clean.
If it's not then it's an error. Also asserts that `./vendor` exists and if not errors.

This required extracting the cache logic from the go command into the
job itself.

[W-6710181](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B00000076rAOIAY/view)

This is currently deployed as `dev:0.0.6` if anyone wants to try it it out.

### CODE REVIEW

**Comment by ghost** [comment]
Example of a failure: https://circleci.com/gh/heroku/go-analysis/120

File: src/golang/jobs/test-nodb.yml [hunk 1]
```diff
@@ -45,11 +53,65 @@ steps:
...
+              if [ -d ./vendor ]; then
+                go mod vendor
+              else
+                echo
+                echo -e '\033[1;93m./vendor does not exist, but "check-vendor" is enabled. Disable check-vendor or populate a vendor folder\e[0m'
+                echo
+                exit 1
+              fi
+              if [ ! -z "$(git status --porcelain)" ]; then
+                git status
```
Comment (mikehale) [inline src/golang/jobs/test-nodb.yml:97]: I noticed that you run `git status` before the error message, but for `Check Modules` you run `git status` afterwards, was that intentional?

File: src/golang/jobs/test-nodb.yml [hunk 2]
```diff
@@ -45,11 +53,65 @@ steps:
...
+            opts: mod verify
+        - go:
+            pkg-spec: ""
+            opts: mod tidy
+        - run:
+            name: Check Modules
+            command: |
+              if [ ! -z "$(git status --porcelain)" ]; then
+                echo
+                echo -e '\033[1;93m!!! "go mod verify && go mod tidy" resulted in changes. Please run those commands locally and commit the changes.\e[0m'
```
Comment (mikehale) [inline src/golang/jobs/test-nodb.yml:78]: It doesn't look like the error you linked actually does anything with the formatting codes. Were you seeing something different? If not maybe they can be removed.

Comment (ghost) [reply src/golang/jobs/test-nodb.yml:78]: it highlights the message as yellow for me.

Comment (ghost) [reply src/golang/jobs/test-nodb.yml:78]: Screen Shot 2019-10-24 at 6 51 31 PM

Comment (mikehale) [reply src/golang/jobs/test-nodb.yml:78]: oh, I think I was just looking at the output on circle-ci, I guess if it is run locally it would highlight. :+1:

**Review by mikehale — APPROVED** [review]
LGTM

**Review by jntullo — APPROVED** [review]
LGTM, thanks!

**Pull request merged**